frontend/src/metabase/query_builder/components/dataref/TableInfoLoader.tsx

```diff
@@ -30,12 +30,13 @@ function useDependentTableMetadata({
   const shouldFetchMetadata = isMissingFields || isMissingFks;
   const [hasFetchedMetadata, setHasFetchedMetadata] =
     useState(!shouldFetchMetadata);
-  const fetchDependentData = useSafeAsyncFunction(() => {
+  const tableId = table.id;
+  const [, fetchDependentData] = useAsyncFn(() => {
     return Promise.all([
-      isMissingFields && fetchMetadata({ id: table.id }),
-      isMissingFks && fetchForeignKeys({ id: table.id }),
+      isMissingFields && fetchMetadata({ id: tableId }),
+      isMissingFks && fetchForeignKeys({ id: tableId }),
     ]);
-  }, [fetchMetadata, table, isMissingFks, isMissingFields, fetchForeignKeys]);
+  }, [fetchMetadata, tableId, isMissingFks, isMissingFields, fetchForeignKeys]);
```

Comment:
This is the fix - `table` was different on every render. `tableId` is a reliable dep.

Reply:
Nit: I think this also works without introducing new variable.
```suggestion
  }, [fetchMetadata, table.id, isMissingFks, isMissingFields, fetchForeignKeys]);
```

Reply:
But then prettier turns this 1 line into 7 lines :see_no_evil: